test/client/api.js

```diff
@@ -82,6 +82,50 @@ module.exports = config => {
     return d.promise
   }
 
+  ClientApi.prototype.doRequestWithToken = function (method, url, token, payload, headers) {
```

Comment:
Naming nit: `WithBearerToken` might make it slightly easier to understand why this is a different method and the others.

Reply:
good idea!